lib/secretsManager.js

```diff
@@ -1,77 +1,169 @@
-let DUMMY_UUID_GEN = 0
-let DUMMY_STORE = []
-
-/**
- * Find all secrets, applying a selector if provided.
- * When provided, all where clause must be fulfilled (logical AND).
- * By default, returns all secrets
- * @async
- * @param {Object} where - hash of where clauses returned secrets must fulfill with
- * @returns {Array<Object>} list (may be empty) of matching secrets
- */
-export async function find(where = {}) {
-  let secrets = []
-
-  try {
-    const props = Object.keys(where)
-    secrets = DUMMY_STORE.filter(secret =>
-      props.every(prop => secret[prop] === where[prop])
+import { v4 as uuid } from 'uuid'
+import aesjs from 'aes-js'
+import * as SecureStore from 'expo-secure-store'
+import AsyncStorage from '@react-native-async-storage/async-storage'
+
+const encryptionPassphraseStorageKey = 'encryptionPassphrase'
+const secretStorageKey = 'secrets'
+
+export default class SecretsManager {
+  constructor() {
+    this.encryptionPassphrase = null
+  }
+
+  async initialize() {
+    let getEncryptionKey = AsyncStorage.getItem
+    let setEncryptionKey = AsyncStorage.setItem
+
+    if (SecureStore.isAvailableAsync()) {
+      getEncryptionKey = SecureStore.getItemAsync
+      setEncryptionKey = SecureStore.setItemAsync
+    }
+
+    const encryptionPassphrase = await getEncryptionKey(
```

Comment:
nit: naming consistency. if the function is called getEncryptionKey, why are we assigning it to a variable called encryptionPassphrase? Is it a key or a passphrase?

Reply:
Good catch! To be frank I was very confused about how to name the variables and functions related with the encryption key. While some third-party libraries I tried called the encryption key, passphrase, others like aes-js called it key. Since we already had the variable storing the storage key value using the "key" word, I then decided to call the thing passphrase to avoid having a variable like "encryptionKeyStorageKey". I genuinely still don't know if there is a difference between using key or passphrase but for now I'll settle with passphrase and change the function name here.